benchmarks/kn-performance/src/commonMain/kotlin/Benchmarks.kt

```diff
@@ -0,0 +1,133 @@
+    val frameBudget: Duration,
+    val percentileCPUAverage: List<BenchmarkPercentileAverage>,
+    val percentileGPUAverage: List<BenchmarkPercentileAverage>,
+    val missedFramesCount: Int,
+    val missedFramesRatio: Double
+) {
+    fun prettyPrint() {
+        percentileCPUAverage.prettyPrint(BenchmarkFrameTimeKind.CPU)
+        percentileGPUAverage.prettyPrint(BenchmarkFrameTimeKind.GPU)
+        println(
+            """                
+            Missed frames count: $missedFramesCount
+            Missed frames ratio: $missedFramesRatio
+            """.trimIndent()
+        )
+    }
+
+    private fun List<BenchmarkPercentileAverage>.prettyPrint(kind: BenchmarkFrameTimeKind) {
+        forEach {
+            println("p${(it.percentile * 100).roundToInt()} ${kind.toPrettyPrintString()} average: ${it.average}")
+        }
+    }
+}
+
+class BenchmarkResult(
+    private val frameBudget: Duration,
+    private val frames: List<BenchmarkFrame>,
+) {
+    private fun percentileAverageFrameTime(percentile: Double, kind: BenchmarkFrameTimeKind): Duration {
+        require(percentile in 0.0..1.0)
+
+        val startIndex = ((frames.size - 1) * percentile).roundToInt()
+
+        val percentileFrames = frames.sortedBy { it.duration(kind) }.subList(startIndex, frames.size)
+
+        return averageDuration(percentileFrames) {
+            it.duration(kind)
+        }
+    }
+
+    fun generateStats(): BenchmarkStats {
+        val missedFramesCount = frames.count {
+            it.cpuDuration + it.gpuDuration > frameBudget
+        }
+
+        return BenchmarkStats(
+            frameBudget,
+            listOf(0.0, 0.5, 0.75, 0.9, 0.95, 0.97, 0.99).map { percentile ->
+                val average = percentileAverageFrameTime(percentile, BenchmarkFrameTimeKind.CPU)
+
+                BenchmarkPercentileAverage(percentile, average)
+            },
+            listOf(0.0, 0.5, 0.95).map { percentile ->
+                val average = percentileAverageFrameTime(percentile, BenchmarkFrameTimeKind.GPU)
+
+                BenchmarkPercentileAverage(percentile, average)
+            },
+            missedFramesCount,
+            missedFramesCount.toDouble() / frames.size
+        )
+    }
+
+    private fun averageDuration(frames: List<BenchmarkFrame>, selector: (BenchmarkFrame) -> Duration): Duration =
+        frames.fold(Duration.ZERO) { acc, frame ->
+            acc + selector(frame)
+        } / frames.size
+
+}
+
+fun runBenchmark(
+    name: String,
+    width: Int,
+    height: Int,
+    targetFps: Int,
+    frameCount: Int,
+    graphicsContext: GraphicsContext?,
+    content: @Composable () -> Unit
+): BenchmarkStats {
+    val stats = measureComposable(frameCount, width, height, targetFps, graphicsContext, content).generateStats()
+
+    println(name)
+    stats.prettyPrint()
+    println()
+
+    return stats
+}
+
+fun runBenchmarks(
+    width: Int = 1920,
+    height: Int = 1080,
+    targetFps: Int = 60,
```

Comment:
Please change default to 120 as we see it is more imortnat to target 120Hz devices